### DISCUSSION TITLE
MVTLayer renderSubLayers Composite Layer issue

### DISCUSSION THREAD

**Discussion by JacobSolomonUCF** [visgl/deck.gl#5827] [Q&A] [answered]
Hi,

I have implemented a custom Composite layer pretty similar to the `LabeledIconLayer` example from the [docs](https://deck.gl/docs/developer-guide/custom-layers/composite-layers#implementing-a-composite-layer). I am returning this layer inside an MVTLayer's `renderSubLayers`. 

See the reproducible [example](https://codesandbox.io/s/deck-gl-composite-point-label-layer-u0jxs)

I can't trigger this behavior 100% of the time, but I can normally trigger it by quickly adjusting the zoom level (via mouse wheel).
![Screen Shot 2021-05-28 at 1 20 09 PM](https://user-images.githubusercontent.com/19891051/120027906-984d7480-bfb9-11eb-9573-b6e441f431a2.png)

My current thought is that, somehow, it is displaying multiple zoom levels at the same time. Changing the `refinementStrategy`, and restricting the `maxZoom` help but doesn't fully resolve this.

Just trying to figure out if there is a bug, or am I missing something?

Thanks!

**Comment by Pessimistress** [comment, marked as answer]
TileLayer/MVTLayer toggles tile visibility by setting the `visible` prop. Your custom layer's sublayers need to inherit the parent layer's visibility, like so:

```diff
- visible: this.props.labelVisible
+ visible: this.props.visible && this.props.labelVisible
```

**Reply by JacobSolomonUCF** [reply]
Ohhh I see. Whew, I can't believe I missed that.

Appreciate your timely reply & support!
